### PR TITLE
Issue#1258 hooks

### PR DESCRIPTION
**Pull request by ioaNikas**
Using hooks in template and material services.

issue #1258

### CODE REVIEW

File: libs/material/src/lib/material/+state/material.service.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { TemplateQuery } from '../../template/+state/template.query';
 import { CollectionConfig, CollectionService } from 'akita-ng-fire';
 import { MaterialState, MaterialStore } from './material.store';
 import { MaterialQuery } from './material.query';
+import { MaterialDocument } from './material.firestore';
```
Comment (Maxstgt) [inline libs/material/src/lib/material/+state/material.service.ts]: shouldn't you export a type from the model first?

File: libs/material/src/lib/template/pages/template-editable/template-editable.component.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import { ChangeDetectionStrategy, Component, HostBinding, OnInit } from '@angular/core';
+import { MaterialControl, MaterialForm } from '../../forms/material.form';
+import { MaterialQuery } from '../../../material/+state/material.query';
+import { MaterialService } from '../../../material/+state/material.service';
+import { MaterialTemplate, createMaterialTemplate } from '../../../material/+state/material.model';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { FormEntity } from '@blockframes/utils/form/forms/entity.form';
+import { MatSnackBar } from '@angular/material/snack-bar';
 import { Observable } from 'rxjs';
+import { Template } from '../../+state/template.model';
```
Comment (Maxstgt) [inline libs/material/src/lib/template/pages/template-editable/template-editable.component.ts]: maybe you want to sort it that everybody can easily read whats being used.

File: libs/movie/src/lib/movie/+state/movie.service.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { Injectable } from '@angular/core';
 import { Movie, createMovie, MovieSale } from './movie.model';
-import { PermissionsService, OrganizationQuery, Organization } from '@blockframes/organization';
+import { OrganizationQuery, Organization } from '@blockframes/organization';
 import { MovieStore, MovieState } from './movie.store';
-import { CollectionService, CollectionConfig } from 'akita-ng-fire';
+import { CollectionService, CollectionConfig, WriteOptions } from 'akita-ng-fire';
 import { switchMap } from 'rxjs/operators';
 import { AngularFirestoreDocument } from '@angular/fire/firestore/document/document';
 import { AngularFirestoreCollection } from '@angular/fire/firestore/collection/collection';
```
Comment (Maxstgt) [inline libs/movie/src/lib/movie/+state/movie.service.ts]: maybe same as above

File: libs/material/src/lib/material/+state/material.service.ts (hunk 1 above)
Comment (ioaNikas) [reply libs/material/src/lib/material/+state/material.service.ts]: Indeed !

**Pull request merged**